scripts/schema-drift.ts

```diff
@@ -0,0 +1,378 @@
+const { promises: fs, createReadStream } = require("fs")
+const path = require("path")
+const { createHash } = require("crypto")
+const { spawn } = require("child_process")
+
+const schemaV1Path = path.join(process.cwd(), "src/schema/v1")
+const schemaV2Path = path.join(process.cwd(), "src/schema/v2")
+
+/** Represents the git status of a file */
+enum FileStatus {
+  Deleted = "deleted",
+  Modified = "modified",
+  Added = "added",
+  Renamed = "renamed",
+  Unknown = "unknown",
+}
+
+type DeltaFileMap = { [filePath: string]: FileStatus }
+
+/**
+ * Converts git status short codes to their `FileStatus` equivalent.
+ * @param change Single character shorthand git file state
+ */
+const GitChangeToFileStatus = (change: string) => {
+  switch (change) {
+    case "M":
+      return FileStatus.Modified
+    case "A":
+      return FileStatus.Added
+    case "D":
+      return FileStatus.Deleted
+    case "R":
+      return FileStatus.Renamed
+    default:
+      return FileStatus.Unknown
+  }
+}
+
+interface BranchState {
+  commitsAhead: number
+  commitsBehind: number
+}
+
+/**
+ * Gives status information on the current branch as compared to origin/master
+ */
+const getBranchDrift = (): Promise<BranchState> =>
+  new Promise((resolve, reject) => {
+    let output = ""
+    const delta = spawn("git", [
+      "rev-list",
+      "--left-right",
+      "--count",
+      "origin/master...HEAD",
+    ])
+
+    delta.stdout.on("data", data => {
+      output += data
+    })
+
+    delta.on("close", code => {
+      if (code !== 0) {
+        reject("Failed to get branch drift")
+      } else {
+        const commitChanges = output.match(/(\d+)\s+(\d+)/)
+        if (!commitChanges) {
+          reject("Something was wrong with the branch drift output")
+        }
+
+        let [, commitsBehind, commitsAhead] = Array.from(commitChanges!).map(
```

Comment:
Ooh, what does that initial comma do? Does it make you select the remainder from the tail of the list?

Reply:
Ah ok, it just skips one. Can you have consecutive commas?